### PR TITLE
Fix to Killcam not appearing

### PR DESCRIPTION
**Pull request by ChrSacher**
Fix various mistakes in the spectator killcam scripts to make the killcam appear again

### CODE REVIEW

**Comment by PiZZAD0X** [comment]
We should use the ace last hit data instead of our own EH

File: uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf [hunk 1]
```diff
@@ -33,46 +33,46 @@ switch (side player) do {
...
+    ESETMVAR(Spectator,Killcam_killer_pos,(ASLtoAGL (eyePos _instigator)));
+} else {
+    //we will try to retrieve info from our hit EH
+    LOG("using hit EH");
+    private _last_hit_info = EGETMVAR(Spectator,Killcam_LastHit,[]);
+    //hit info retrieved, now we check if it's not caused by fall damage etc.
+    //also we won't use info that's over 10 seconds old
+    if !(_last_hit_info isEqualto []) then {
+        _last_hit_info params ["_data", "_time", "_unitPos", "_killerPos"];
+        if ((_time + 10 > time) &&
```
Comment (PiZZAD0X) [inline uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf]: `if ((_time + 10 > CBA_MissionTime) &&`

File: uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf [hunk 2]
```diff
@@ -33,46 +33,46 @@ switch (side player) do {
     [_unit] joinSilent grpNull;
 }, [_unit], 2] call CBA_fnc_WaitAndExecute;
 
-if (EGETMVAR(Spectator,Killcam_Enabled,true)) then {
```
Comment (PiZZAD0X) [inline uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf]: Yeah we can assume everyone is going to use killcam and instant death, have these settings been removed from the 3den menus?

Comment (ChrSacher) [reply uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf]: Spectator,Killcam_Enabled does not exist in the anymore. You mad it factionized to Killcam_Enabled_Blufor.

Comment (PiZZAD0X) [reply uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf]: Then those faction settings don't have an effect and don't need to be in those menus. Can you consider killcam ever being needing to be disabled?

Comment (ChrSacher) [reply uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf]: I personally would leave the option in. You never know what someone might want and it doesn't have any major impact.

File: uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf [hunk 3]
```diff
@@ -33,23 +33,30 @@ switch (side player) do {
...
+
+//we check if player didn't kill himself or died for unknown reasons
+if ((vehicle _instigator != vehicle _unit) && {!isNull _instigator}) then {
+    //this is the standard case (killed EH got triggered by getting shot)
+    LOG("Getting killer by killed EH");
+    ESETMVAR(Spectator,Killcam_unit_pos,(ASLtoAGL (eyePos _unit)));
+    ESETMVAR(Spectator,Killcam_killer,_instigator);
+    ESETMVAR(Spectator,Killcam_killer_pos,(ASLtoAGL (eyePos _instigator)));
+} else {
+    private _aceLastDamage = _unit getVariable ["ace_medical_lastDamageSource", objNull];
```
Comment (PiZZAD0X) [inline uo_sys_framework/Core/cfgFunctions/EventHandlers/fn_KilledHandler.sqf]: Perfect, ace EH in addition to ours

**Pull request merged**